src/workerd/io/worker-entrypoint.c++

```diff
@@ -310,14 +317,23 @@ kj::Promise<void> WorkerEntrypoint::request(kj::HttpMethod method,
                          ->getSignal());
     }
 
-    return lock.getGlobalScope().request(method, url, headers, requestBody, wrappedResponse,
-        cfBlobJson, lock,
-        lock.getExportedHandler(entrypointName, kj::mv(props), context.getActor()), kj::mv(signal));
-  })
-      .then([this](api::DeferredProxy<void> deferredProxy) {
-    TRACE_EVENT("workerd", "WorkerEntrypoint::request() deferred proxy step",
-        PERFETTO_FLOW_FROM_POINTER(this));
-    proxyTask = kj::mv(deferredProxy.proxyTask);
+    return lock.getGlobalScope()
+        .request(method, url, headers, requestBody, wrappedResponse, cfBlobJson, lock,
+            lock.getExportedHandler(entrypointName, kj::mv(props), context.getActor()),
+            kj::mv(signal))
+        .then([this, &context, &wrappedResponse](api::DeferredProxy<void> deferredProxy) {
+      TRACE_EVENT("workerd", "WorkerEntrypoint::request() deferred proxy step",
+          PERFETTO_FLOW_FROM_POINTER(this));
+      proxyTask = kj::mv(deferredProxy.proxyTask);
+      KJ_IF_SOME(t, context.getWorkerTracer()) {
```

Comment:
I moved the deferred proxy promise continuation under the context.run block here, otherwise there is a segfault (perhaps due to context no longer being valid). Open to alternative suggestions.